repose-aggregator/components/filters/openapi-validator-filter/src/main/scala/org/openrepose/filters/openapivalidator/OpenApiValidatorFilter.scala

```diff
@@ -0,0 +1,205 @@
+            httpResponse.sendError(validationFailure.issue.statusCode, validationFailure.message.getMessage)
+          case None =>
+            logger.trace("Successfully validated request -- continuing processing")
+            chain.doFilter(bufferingHttpServletRequest, httpResponse)
+        }
+      case Failure(e: RequestConversionException) =>
+        logger.error("Failed to convert the servlet request to a validation request", e)
+        httpResponse.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getMessage)
+      case Failure(e) =>
+        val statusCode = HttpServletResponse.SC_INTERNAL_SERVER_ERROR
+        logger.error("Failed to validate request -- rejecting with status code: '{}' for reason '{}'", statusCode.toString, e.getMessage, e)
+        httpResponse.sendError(statusCode, e.getMessage)
+    }
+  }
+
+  override def doConfigurationUpdated(newConfiguration: OpenApiValidatorConfig): OpenApiValidatorConfig = {
+    validator = OpenApiInteractionValidator
+      .createFor(resolveHref(newConfiguration.getHref))
+      .build()
+
+    newConfiguration
+  }
+
+  /**
+    * Resolves relative [[URI]] representations as files relative to the configuration root directory.
+    *
+    * @param href a [[String]] representation of a [[URI]]
+    * @return a [[String]] representation of an absolute [[URI]]
+    */
+  private def resolveHref(href: String): String = {
+    val hrefUri = URI.create(href)
+    if (hrefUri.isAbsolute) {
+      // The URI is absolute, so return it as-is.
+      // This handles hrefs pointing to remote resources (e.g., HTTP, FTP).
+      hrefUri.toString
+    } else {
+      // The URI is not absolute, so process it as a file.
+      val hrefFile = new File(href)
+      if (hrefFile.isAbsolute) {
+        // The file path is absolute, so return the absolute URI for the file path.
+        hrefFile.getAbsolutePath
+      } else {
+        // The file path is relative, so resolve it relative to the configuration directory
+        // and return the path.
+        new File(configurationRoot, href).toURI.toString
+      }
+    }
+  }
+}
+
+object OpenApiValidatorFilter {
+
+  // A mapping from Message keys defined by the validation library to data defining how we will handle those Messages.
+  // Note that the path and method checks necessarily occur before all other checks since they are required
+  // to resolve the API Operation in the OpenAPI document.
+  // If either one fails, no further checks will be performed.
+  // As such, the validation order will be (path -> method -> everything else).
+  // For that reason, path and method issues are given the highest and second highest priorities respectively.
+  final val ValidationIssues: Map[String, ValidationIssue] = Map(
+    "validation.request.path.missing" -> ValidationIssue(HttpServletResponse.SC_NOT_FOUND, -2),
```

Comment:
Well, we can only respond with one status code for any given request. Since different validation issues may be mapped to different status codes, and since multiple validation issues can be reported for any given request, priorities are a way to:

- Be consistent about which issue to respond to.
  - We are not strictly guaranteed that the report messages will be in any particular order. However, since `List`s are used in the current `ValidationReport` implementation, the order should actually be consistent. However, the implementation could change. This is a way to be explicit about ordering regardless of what the underlying library does.
- Dictate which issues are more important to respond to. For example, if we have two issues, one which would result in a `415` response and another which would result in a `500` response, we may prefer to respond with the `415` over the `500` since it is more specific and is actionable.

As for negative numbers, that's fine, they can really be whatever. The reason I went with negative numbers was to indicate that those two issues are handled differently than all of the other issues. I meant to convey that the path will be validated first, and on failure, will be the only failure. Then the method will be validated, and on failure, will be the only failure. Then all of the other parts of the request will be validated, and there may be multiple failures.

As for going with consecutive numbers, I'm not sure that really matters. The numbers are all internal, so having to redo them is a relatively small pain that is only exposed to the operator if ordering of existing issues changes. That being said, I can add gaps if you'd like -- it might save a handful of seconds whenever (if ever) we have to add a new issue. How far apart would you space them?

Reply:
Actually, I think I have a better way of doing this. I never liked using numbered priorities since they seem kind of arbitrary, but I couldn't figure out a better way to encode priority. Until maybe now. I'll just add the issues to an ordered collection (maybe just a `List` since this probably won't need to be a `Map` anymore, though a `LinkedHashMap` may be alright to avoid needing the `case class`) and use the order to represent priority. Then I'll change the algorithm to get the highest priority issue to iterate through our sorted (prioritized) issue map and find the first issue that matches an issue from the report. To keep things efficient, I'll probably transform the report messages into a `HashMap` as well so that I can do a quick lookup (`O(n) + O(m)`) rather than iterating through both collections (`O(nm)`).

Reply:
Yeah, i mentioned the consecutive numbers thing before i saw usage.